storage/keyspace/raw_iterator.rs

```diff
@@ -66,6 +79,7 @@ impl Seekable<[u8]> for DBIterator {
                 Ordering::Less => {
                     self.item.take();
                     self.iterator.seek(key);
+                    self.item = peek_item(&self.iterator); // repopulate `item` to prevent advancing the underlying iterator
```

Comment:
Bug fix! When item is empty, that means that the kv pair the Rocks iterator _is currently pointing to_  has been yielded to the user, and the underlying iterator needs to be advanced before peeking. That means that after a seek, we always attempted to advance the iterator once more.